### PR TITLE
Adding node sample for thermostat

### PR DESCRIPTION
**Pull request by rido-min**
This PR adds a node device simulator implementing the Thermostat model.

It's using the latest NPM Package

/cc @elhorton @jebrando @anthonyvercolano

### CODE REVIEW

File: node/thermostat/main.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+const os = require('os')
+
+const DigitalTwinClient = require('azure-iot-digitaltwins-device').DigitalTwinClient
+
+const DeviceInformation = require('./deviceInformation').DeviceInformation
+const SdkInformation = require('./sdkInformation').SdkInformation
+const TemperatureSensor = require('./temperatureSensor').TemperatureSensor
+const Diagnostics = require('./diagnostics').Diagnostics
+
+const modelId = 'dtmi:com:example:Thermostat;1'
```
Comment (anthonyvercolano) [inline node/thermostat/main.js:12]: >modelId [](start = 6, length = 7)

You don't include a model file in your example.

File: node/thermostat/main.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  totalStorage: 65000,
+  totalMemory: os.totalmem()
+}
+
+const sleep = (ms) => {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+const adjustTemp = async (target) => {
+  const step = (parseFloat(target) - currentTemp) / 10
```
Comment (anthonyvercolano) [inline node/thermostat/main.js:79]: >target [](start = 27, length = 6)

Wasn't target already a float?

File: node/thermostat/main.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    await sleep(1000)
+  }
+}
+
+const startTelemetryLoop = async () => {
+  telemetryLoop = setInterval(async () => {
+    if (currentTemp !== targetTemp) {
+      console.log('sending temp ' + currentTemp)
+      await digitalTwinClient.sendTelemetry(tempSensor, { temperature: currentTemp })
+      digitalTwinClient.report(tempSensor, { currentTemperature: currentTemp })
```
Comment (anthonyvercolano) [inline node/thermostat/main.js:92]: >digitalTwinClient.report [](start = 6, length = 24)

Two things.

1) shouldn't there be an await on the report?
2) I generally don't think of reports being something as frequent as a telemetry.  But hey, it's your device.

File: node/thermostat/main.js (hunk 1 above)
Comment (ridomin) [reply node/thermostat/main.js:12]: the model is in the target branch. Mode First :)

File: node/thermostat/main.js (hunk 2 above)
Comment (ridomin) [reply node/thermostat/main.js:79]: good catch, I guess the `parseFloat` is not needed. Updating

File: node/thermostat/main.js (hunk 3 above)
Comment (ridomin) [reply node/thermostat/main.js:92]: 1. Yes, but `.report` does not return a `Promise`. Is this a bug?
2. Yeah, I know that sending the same value as telemetry and a property has a reason. (Following someone else pattern)

**Pull request merged**